packages/cli/src/dbt/models.ts

```diff
@@ -68,27 +69,51 @@ type GenerateModelYamlArgs = {
     model: CompiledModel;
     table: WarehouseTableSchema;
     includeMeta: boolean;
+    dbtVersion?: SupportedDbtVersions;
 };
 const generateModelYml = ({
     model,
     table,
     includeMeta,
-}: GenerateModelYamlArgs) => ({
-    name: model.name,
-    columns: Object.entries(table).map(([columnName, dimensionType]) => ({
-        name: columnName,
-        description: '',
-        ...(includeMeta
-            ? {
-                  meta: {
-                      dimension: {
-                          type: dimensionType,
-                      },
-                  },
-              }
-            : {}),
-    })),
-});
+    dbtVersion,
+}: GenerateModelYamlArgs) => {
+    // Check if we should use dbt 1.10+ metadata structure
+    const useDbt110Metadata =
+        dbtVersion &&
+        Object.values(SupportedDbtVersions).indexOf(dbtVersion) >=
```

Comment:
We're making this comparison in a couple places now. A function or reworking the enum would might make the operation clearer.